### PR TITLE
Copy Waste: Add SharedAddress component with address copying

### PR DESCRIPTION
**Pull request by Shadowfiend**
Add `SharedAddress` component and use it in transaction signing
and activity details.

Current limitations:
- Tooltip is an OS tooltip using `title`, preferences may vary.
- We don't propagate name system information and don't yet have
  a design for how to tell the user which name system was used
  to determine the name they see.
- Names are not resolved in as many places as might be ideal.
- There is no icon indicator that a copy is possible.

That said, anywhere a name is resolved it is displayed, tooltip
has the address, and clicking copies, so the basis for the
functionality we want is there and we are in a strictly better
place with these changes.

Screenshots:

Screen Shot 2022-03-22 at 21 57 10

Screen Shot 2022-03-22 at 21 57 30

See #649.

### CODE REVIEW

**Comment by Shadowfiend** [comment]
Rebased on main, added hover color, and added extra info in tooltip.

File: ui/components/Shared/SharedAddress.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ *
+ * If the optional `nameResolverSystem` property is provided, an info tooltip
+ * is included in the component to inform the user of which name resolver
+ * system was used to resolve the passed `name`. If no `name` is passed, the
+ * `nameResolverSystem` prop is ignored.
+ */
+export default function SharedAddress({
+  name,
+  address,
+  nameResolverSystem: nameSourceSystem,
```
Comment (0xDaedalus) [inline ui/components/Shared/SharedAddress.tsx:32]: Just curious why use `nameSourceSystem` inside of the component but `nameResolverSystem` outside.

**Review by 0xDaedalus — DISMISSED** [review]
Tested and works great!  I do want to mention though that the user experience of hovering over text - seeing it turn gold - cursor turning into a pointer - and the resulting interaction from clicking NOT being a link is a bit jarring (my mind wants it to be a link that opens in etherscan after seeing gold hover).  Not at all blocking but just wanted to share my subjective experience.

File: ui/components/Shared/SharedAddress.tsx (hunk 1 above)
Comment (Shadowfiend) [reply ui/components/Shared/SharedAddress.tsx:32]: This was me missing a sloppy auto-rename.

**Comment by Shadowfiend** [comment]
Fixed all those issues :point_up_2:

**Review by 0xDaedalus — APPROVED** [review]

**Comment by Shadowfiend** [comment]
> I do want to mention though that the user experience of hovering over text - seeing it turn gold - cursor turning into a pointer - and the resulting interaction from clicking NOT being a link is a bit jarring (my mind wants it to be a link that opens in etherscan after seeing gold hover).

Talk to the @mhluongo s if the world lol. I absolutely hate it but it's how most implementations work due to the pervasiveness of the action.

**Comment by mhluongo** [comment]
Just speaking up for our users :stuck_out_tongue:

**Pull request merged**